src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -231,6 +232,19 @@ else if (shouldStartLine && !isOnStartOfLine(details, targetSourceLine)) {
     }
 
     /**
+     * Checks that a right curly should be alone on line or alone with SEMI token is LAMBDA.
+     * @param bracePolicy option for placing the right curly brace
+     * @param details Details for validation
+     * @return true if a right curly should be alone on line or with semi if token is LAMBDA.
+     */
+    private static boolean shouldBeAlone(RightCurlyOption bracePolicy, Details details) {
```

Comment:
I don't like this name and logic inside 'shouldBeAlone'. I agree that adding additional boolean conditions into shouldBeAloneOnLine will make the method too big and hard to understand, but maybe it is a reason to split it or refactor.